discord/utils.py

```diff
@@ -1027,3 +1029,53 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
     if style is None:
         return f'<t:{int(dt.timestamp())}>'
     return f'<t:{int(dt.timestamp())}:{style}>'
+
+
+def basic_autocomplete(values: Union[Iterable[str],
+                                     Callable[[Interaction], Union[Iterable[str], Coroutine[Iterable[str]]]],
+                                     Coroutine[Iterable[str]]]) -> Callable[[Interaction, str], Coroutine[List[str]]]:
+    """A helper function to make a basic autocomplete for slash commands. This is a pretty standard autocomplete and
+    will return any options that start with the value from the user, case insensitive. If :param:`values` is callable,
+    it will be called with the interaction.
+
+    This is meant to be passed into the :attr:`discord.Option.autocomplete` attribute.
+
+    Example
+    --------
+
+    .. code-block:: python3
+
+        Option(str, "color", autocomplete=basic_autocomplete(("red", "green", "blue")))
+
+        # or
+
+        async def autocomplete(interaction):
+            return ("foo", "bar", "baz", interaction.user.name)
+
+        Option(str, "name", autocomplete=basic_autocomplete(autocomplete))
+
+
+    .. versionadded:: 2.0
+
+    Parameters
+    -----------
+    values: Union[Iterable[:class:`str`], Callable[[:class:`Interaction`], Union[Iterable[:class:`str`], Coroutine[Iterable[:class:`str`]]]], Coroutine[Iterable[:class:`str`]]]
+        Possible values for the option. Accepts an iterable of :class:`str`, a callable (sync or async) that takes a
+        single argument of :class:`Interaction`, or a coroutine. Must resolve to an iterable of :class:`str`.
+
+    Returns
+    --------
+    Callable[[:class:`Interaction`, :class:`str`], Coroutine[List[:class:`str`]]]
+        A wrapped callback for the autocomplete.
+    """
+    async def autocomplete_callback(interaction: Interaction,
+                                    value: str) -> List[str]:
+        _values = values  # since we reassign later, python considers it local if we don't do this
+
+        if callable(_values):
+            _values = _values(interaction)
+        if asyncio.iscoroutine(_values):
+            _values = await _values
+        return [x for x in _values if x.lower().startswith(value.lower())]
+
+    return autocomplete_callback
```

Comment:
An alternative implementation, more similar to [when_mentioned_or](https://pycord.readthedocs.io/en/master/ext/commands/api.html#discord.ext.commands.when_mentioned_or). Of course, the functionality here would also be usable in the implementation provided in the original code, aside from passing an iterable instead of variadic arguments.
```suggestion


def basic_autocomplete(*values: str) -> Callable[[Interaction, str], Coroutine[List[str]]]:
    """A helper function to make a basic autocomplete for slash commands. This is a pretty standard autocomplete and
    will return any options that start with the value from the user, case insensitive.
    These are meant to be passed into the :attr:`discord.Option.autocomplete` attribute.
    
    Example
    --------
    
    .. code-block:: python3
    
        Option(str, "color", autocomplete=basic_autocomplete(*("red", "green", "blue")))
        
    .. note::
    
        This callable returns another callable, so if this is done inside a custom
        callable, you must call the returned callable, for example:
        
        .. code-block:: python3
        
            async def autocomplete(interaction):
                choices = "foo", "bar", "baz", interaction.user.name
                return await basic_autocomplete(*choices)(interaction)
            
            
    .. versionadded:: 2.0
    
    Parameters
    -----------
    values: :class:`str`
        Possible values for the option. 
        
    Returns
    --------
    Callable[[:class:`Interaction`, :class:`str`], Coroutine[List[:class:`str`]]]
        A wrapped callback for the autocomplete.
    """
    async def autocomplete_callback(interaction: Interaction,
                                    value: str) -> List[str]:
        return [x for x in values if x.lower().startswith(value.lower())]
    return autocomplete_callback
```

Reply:
Using the second method would probably be simpler to use comparing
```python
basic_autocomplete(["red", "yellow"])
```
to
```python
basic_autocomplete("red", "yellow")
```